lib/nypl_sierra_api_client.rb

```diff
@@ -55,49 +57,66 @@ def do_request (method, path, options = {})
 
     authenticate! if options[:authenticated]
 
-    uri = URI.parse("#{@config[:base_url]}#{path}")
-
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = uri.scheme === 'https'
+    @uri = URI.parse("#{@config[:base_url]}#{path}")
```

Comment:
This could be reworked to regenerate `uri` when needed. Would it make more sense for `http` (line 81) to be an instance var?

Reply:
My initial concern was about using an instance variable at all. A client serving multiple requests should probably not store details about the most recent request solely for the purpose of passing those details from one method to another; Ideally the client would pass all of the details about the request in the method params themselves. The problem I see with storing instance variables is you risk allowing a previous/current request to interfere with another request. That concern is kind of academic in Ruby, where the risk is arguably zero. (In an event-driven lang like JS, the asyncronicity would make it a problem.) Changing it to be more "functional" in style (passing request details from one method to another to avoid storing instance vars) would be a big reworking of the original client, so I don't think it's worth it. The @uri instance var is a pragmatic choice here.